subsys/Makefile

```diff
@@ -1,3 +1,5 @@
+subdir-ccflags-y += -D__ZEPHYR_KERNEL__
```

Comment:
This one seems dangerous: are we sure none of the subsystems provide library code that works from userspace?  Even if they don't now (by definition, really), they surely will  in the future.  Think of stuff like network protocol implementations for DHCP, DNS, etc...  None of that requires kernel access.  Might be better to force subsystem code to do this per-file.

Reply:
I think doing this per-file would be hard to deal with. In the future I'd like to support use-cases where the application(s) and kernel are separately linked, and share code only in the runtime library and other stuff under lib/.

I think if subsystem code is revised to work in user mode, we should just move it somewhere under lib/.

Reply:
Still think this is OK to do.
I don't think we should have mixed kernel and library code, if we get the network stack to the point where some or all of it can be done in userland, it should be put inside or sibling to the runtime library.